overviewwidget.cpp

```diff
@@ -88,3 +93,20 @@ void OverviewWidget::modelUpdated()
 {
     m_treeView->expandAll();
 }
+
+void OverviewWidget::onItemActivated(const QModelIndex &index)
+{
+    if (!index.isValid())
+        return;
+
+    ModelTreeNode *node = static_cast<ModelTreeNode *>(index.internalPointer());
+
+    // it is assumed, that if item has children, user wants to expand it, even though sometimes it could be jumped to.
```

Comment:
Fixed the issue, so now double click on any item will move to its location in source (if the location is available). To expand tree level, the small triangle which indicates that the item is expandable needs to be selected (Project Tree in MonoDevelop works the same :P ).